### DISCUSSION TITLE
what is raven?

### DISCUSSION THREAD

**Discussion by rickyJ43** [idaholab/raven#2404] [Q&A] [answered]
what is raven? for what? for how? someone have a good idea please answer the question..

**Comment by wangcj05** [comment]
You can find detailed descriptions at https://github.com/idaholab/raven/wiki.

**Comment by Jimmy-INL** [comment, marked as answer]
While it might not be the original purpose of RAVEN, currently RAVEN is a Machine Learning pipeline orchestrator. It can drive the full workflow, from interfacing with external models or codes to sampling these codes, perturbing their inputs, running them in parallel, parsing their outputs, to building sophisticated Reduced Order Models (eg., SVM, GP, Neural Networks, etc.), to performing deep analytics, post-processing, such as sensitivity analyses, and uncertainty quantification, to deaign optimization and economical modeling. RAVEN also is the engine for many stand alone plugins, in case users cannot open source their entire work flow, they can have a unifies licensed plugin. We have several of those from other national labs as well.
